### PR TITLE
Update kibana MITRE version to `v14.1` [classic]

### PR DESCRIPTION
**Pull request by joepeeples**
Contributes to https://github.com/elastic/security-docs/issues/4550 by updating the referenced MITRE version to 14.0 in the classic docs.

> [!IMPORTANT]  
> - Don't merge this until https://github.com/elastic/kibana/pull/174120 has been merged first.
> - Also, we'll probably update the version in serverless several weeks before the Stack in 8.14.0, but it's OK to merge this at the same time, as long as we're just updating `main` (currently == 8.14).

---
- Twin PR: https://github.com/elastic/staging-serverless-security-docs/pull/289

### CODE REVIEW

**Comment by joepeeples** [comment]
The version label on https://github.com/elastic/kibana/pull/174120 has been changed so this change could be delayed further.

File: docs/detections/rules-coverage.asciidoc [hunk 1]
```diff
@@ -12,7 +12,7 @@ Mirroring the MITRE ATT&CK® framework, columns represent major tactics, and cel
 
 [NOTE]
 ====
-This page only includes the detection rules you currently have installed, and only rules that are mapped to MITRE ATT&CK® tactics. Additionally, rules must be mapped to the MITRE ATT&CK® version used by {elastic-sec}: `v13.1`. Elastic prebuilt rules that aren't installed and custom rules that are either unmapped or mapped to a deprecated tactic or technique will not be included.
+This page only includes the detection rules you currently have installed, and only rules that are mapped to MITRE ATT&CK® tactics. Additionally, rules must be mapped to the MITRE ATT&CK® version used by {elastic-sec}: `v14.0`. Elastic prebuilt rules that aren't installed and custom rules that are either unmapped or mapped to a deprecated tactic or technique will not be included.
```
Comment (joepeeples) [inline docs/detections/rules-coverage.asciidoc:15]:
```suggestion
This page only includes the detection rules you currently have installed, and only rules that are mapped to MITRE ATT&CK® tactics. Additionally, rules must be mapped to the MITRE ATT&CK® version used by {elastic-sec}: `v14.1`. Elastic prebuilt rules that aren't installed and custom rules that are either unmapped or mapped to a deprecated tactic or technique will not be included.
```

**Review by dplumlee — DISMISSED** [review]

**Review by benironside — APPROVED** [review]

File: docs/detections/rules-coverage.asciidoc [hunk 2]
```diff
@@ -12,7 +12,7 @@ Mirroring the MITRE ATT&CK® framework, columns represent major tactics, and cel
 
 [NOTE]
 ====
-This page only includes the detection rules you currently have installed, and only rules that are mapped to MITRE ATT&CK® tactics. Additionally, rules must be mapped to the MITRE ATT&CK® version used by {elastic-sec}: `v13.1`. Elastic prebuilt rules that aren't installed and custom rules that are either unmapped or mapped to a deprecated tactic or technique will not be included.
+This page only includes the detection rules you currently have installed, and only rules that are mapped to MITRE ATT&CK®. The coverage page maps detections to the following https://attack.mitre.org/resources/updates/updates-october-2023[MITRE ATT&CK® version] used by {elastic-sec}: `v14.1`. Elastic prebuilt rules that aren't installed and custom rules that are either unmapped or mapped to a deprecated tactic or technique will not appear on the coverage map.
```
Comment (joepeeples) [inline docs/detections/rules-coverage.asciidoc:15]:
```suggestion
This page only includes the detection rules you currently have installed, and only rules that are mapped to MITRE ATT&CK®. The coverage page maps detections to the following https://attack.mitre.org/resources/updates/updates-october-2023[MITRE ATT&CK® version] used by {elastic-sec}: `v14.1`. Elastic prebuilt rules that aren't installed and custom rules that are either unmapped or mapped to a deprecated tactic or technique do not appear on the coverage map.
```

File: docs/detections/rules-coverage.asciidoc [hunk 3]
```diff
@@ -12,7 +12,7 @@ Mirroring the MITRE ATT&CK® framework, columns represent major tactics, and cel
 
 [NOTE]
 ====
-This page only includes the detection rules you currently have installed, and only rules that are mapped to MITRE ATT&CK® tactics. Additionally, rules must be mapped to the MITRE ATT&CK® version used by {elastic-sec}: `v13.1`. Elastic prebuilt rules that aren't installed and custom rules that are either unmapped or mapped to a deprecated tactic or technique will not be included.
+This page only includes the detection rules you currently have installed, and only rules that are mapped to MITRE ATT&CK®. The coverage page maps detections to the following https://attack.mitre.org/resources/updates/updates-october-2023[MITRE ATT&CK® version] used by {elastic-sec}: `v14.1`. Elastic prebuilt rules that aren't installed and custom rules that are either unmapped or mapped to a deprecated tactic or technique do not appear on the coverage map.
```
Comment (joepeeples) [inline docs/detections/rules-coverage.asciidoc:15]:
```suggestion
This page will only include the detection rules you currently have installed, and only rules that are mapped to MITRE ATT&CK®. The coverage page maps detections to the following https://attack.mitre.org/resources/updates/updates-october-2023[MITRE ATT&CK® version] used by {elastic-sec}: `v14.1`. Elastic prebuilt rules that aren't installed and custom rules that are either unmapped or mapped to a deprecated tactic or technique do not appear on the coverage map.
```

**Review by natasha-moore-elastic — APPROVED** [review]

**Pull request merged**